### PR TITLE
docs(security): document Telegram webhook migration

### LINKED ISSUE #3147: Security: Stop using the Telegram bot token as the webhook path
## Description
The Telegram router registers the webhook at `/${botToken}` and documents the token-bearing URL as the security mechanism. Webhook URLs are commonly exposed in proxies, logs, screenshots, and support tools. A leaked path leaks the bot token itself and gives attackers both the secret URL and the credential needed to call Telegram APIs.

## Affected locations
- `packages/franken-orchestrator/src/comms/channels/telegram/telegram-router.ts:14` describes token-in-path security.
- `packages/franken-orchestrator/src/comms/channels/telegram/telegram-router.ts:21` mounts `app.post(`/${botToken}`, ...)`.

## Recommendation
Use Telegram secret_token header validation or a random opaque webhook secret that is not the bot token; redact any token-derived path from logs.

## Acceptance criteria
- Webhook authentication no longer embeds the bot token in the URL.
- Invalid or missing secret headers are rejected.
- Migration docs cover rotating existing webhook URLs.

## LLM handoff notes
Preserve existing behavior for safe development paths, avoid logging secrets or provider response bodies, and add regression tests for the security boundary.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- document the fixed Telegram webhook endpoint and dedicated secret header
- add migration guidance for legacy token-bearing webhook URLs
- add a README regression test for issue #3147 requirements

## Verification
- `npm run test:root -- tests/docs-issue-3147.test.ts`
- `npm run test --workspace @franken/orchestrator -- tests/unit/comms/telegram-router.test.ts tests/unit/http/comms-routes.test.ts`
- `npm run build`
- `npm run typecheck`
- `npm run lint`
- `npm test` (one unrelated timeout in `dep-factory-providers.test.ts`; isolated rerun passed all 48 tests)

Closes #3147

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: README.md [hunk 1]
```diff
@@ -923,11 +923,24 @@ External communications are implemented in `@franken/orchestrator` under `packag
...
 
 Channels route through the orchestrator comms pipeline. See [ADR-016](docs/adr/016-external-comms-gateway.md) for the original gateway decision and the orchestrator comms source for current implementation details.
 
+### Telegram webhook setup and migration
+
+Telegram updates use the fixed `https://<public-host>/webhooks/telegram` route. Configure a dedicated, randomly generated webhook secret through `comms.telegram.webhookSecretTokenRef`, then pass the resolved value as Telegram's `secret_token` when registering the webhook. Telegram sends that value in `X-Telegram-Bot-Api-Secret-Token`; Frankenbeast rejects requests whose header is missing or invalid. Do not append the bot token to the webhook URL.
+
+To migrate an existing token-bearing webhook URL:
+
+1. Generate and store a new webhook secret that is independent of the Telegram bot token, then update `comms.telegram.webhookSecretTokenRef` through `frankenbeast init --repair` or the configured secret backend.
```
[the remark this answers is not in the document]
Comment (djm204) [reply README.md]: Fixed in 4d134c41. The migration now directs operators to update the configured secret backend directly and explicitly warns that `init --repair` exits without rotating secrets when verification already passes.

File: README.md [hunk 2]
```diff
@@ -923,11 +923,24 @@ External communications are implemented in `@franken/orchestrator` under `packag
...
 | Discord | Gateway events | ED25519 signature verification |
-| Telegram | Webhook | Token-based authentication |
+| Telegram | Webhook | Telegram `secret_token` header validation |
 | WhatsApp | Cloud API | SHA256 signature verification |
 
 Channels route through the orchestrator comms pipeline. See [ADR-016](docs/adr/016-external-comms-gateway.md) for the original gateway decision and the orchestrator comms source for current implementation details.
 
+### Telegram webhook setup and migration
+
+Telegram updates use the fixed `https://<public-host>/webhooks/telegram` route. Configure a dedicated, randomly generated webhook secret through `comms.telegram.webhookSecretTokenRef`, then pass the resolved value as Telegram's `secret_token` when registering the webhook. Telegram sends that value in `X-Telegram-Bot-Api-Secret-Token`; Frankenbeast rejects requests whose header is missing or invalid. Do not append the bot token to the webhook URL.
```
[the remark this answers is not in the document]
Comment (djm204) [reply README.md]: Fixed in 4d134c41. The setup section now states Telegram’s 1–256 character limit and the allowed A-Z, a-z, 0-9, underscore, and hyphen character set.

File: README.md [hunk 3]
```diff
@@ -923,11 +923,24 @@ External communications are implemented in `@franken/orchestrator` under `packag
...
+
+To migrate an existing token-bearing webhook URL:
+
+1. Generate and store a new webhook secret that is independent of the Telegram bot token, then update `comms.telegram.webhookSecretTokenRef` through `frankenbeast init --repair` or the configured secret backend.
+2. After storing the secret, re-register the webhook with Telegram using the fixed `/webhooks/telegram` URL and the same new value in the Bot API `secret_token` parameter. Telegram's `setWebhook` call replaces the previous webhook URL.
+3. Send a test update and confirm the fixed route accepts the `X-Telegram-Bot-Api-Secret-Token` header while the old token-bearing path returns `404`.
```
[the remark this answers is not in the document]
Comment (djm204) [reply README.md]: Fixed in 4d134c41. The verification step now scopes 404 behavior to direct chat-server requests and documents dashboard-web’s legacy-path rewrite behavior.

**Comment by djm204** [comment]
@codex review
